src/sys/socket/addr.rs

```diff
@@ -958,30 +779,10 @@ pub mod netlink {
     use ::sys::socket::addr::AddressFamily;
     use libc::{sa_family_t, sockaddr_nl};
     use std::{fmt, mem};
-    use std::hash::{Hash, Hasher};
 
-    #[derive(Copy, Clone)]
+    #[derive(Copy, Clone, Debug, Eq, Hash, PartialEq)]
     pub struct NetlinkAddr(pub sockaddr_nl);
 
-    // , PartialEq, Eq, Debug, Hash
-    impl PartialEq for NetlinkAddr {
```

Comment:
The derived PartialEq and Hash will use the sockaddr_nl.nl_pad field, which is probably not what we want.